lib/suppression_list/list_entry.ex

```diff
@@ -0,0 +1,18 @@
+defmodule SparkPost.SuppressionList.ListEntry do
+  @moduledoc """
+  SparkPost representation of a suppression list entry.
+
+  ## Fields
+   - recipient: Email address that was suppressed
+   - type: Type of suppression record (transactional or non_transactional)
+   - source: Source responsible for inserting the list entry. Can be one of
+      "Spam Complaint", "List Unsubscribe", "Bounce Rule", "Unsubscribe Link",
+      "Manually Added", or "Compliance"
+   - description: explanation of suppression
+  """
+
+  defstruct recipient: :required,
```

Comment:
Although `transactional` and `non_transactional` are deprecated, they can still turn up in existing records so if we plan to use this module to represent received suppression list entries, I suggest we include them too.